chore: add storage policy documentation (#4246)
hashicorp/boundary: website/content/docs/configuration/session-recording/update-storage-policy.mdx

---
layout: docs
page_title: Update storage bucket policies
description: |-
  How to update a storage bucket policy for session recordings in Boundary
---

# Update storage bucket policies

<EnterpriseAlert product="boundary">This feature requires <a href="https://www.hashicorp.com/products/boundary">HCP Boundary or Boundary Enterprise</a></EnterpriseAlert>

A [storage policy][] codifies [storage bucket][] lifecycle management for [session recordings][].

A storage policy exists in either the global scope or an org scope. Storage policies that are created in the global scope can be associated with any org scope.

> When a storage policy is updated, new session recordings within a scope where the policy is applied will conform to the updated policy. Existing recordings will maintain the previous policy unless the new policy is retroactively applied.

Reviewer: ```suggestion
When you update a storage policy, new session recordings within a scope where the policy is applied will conform to the updated policy. Existing recordings will maintain the previous policy unless the new policy is retroactively applied.
```

For more information about setting up storage bucket policies, refer to the [configure storage bucket policies](/boundary/docs/configuration/session-recording/configure-storage-policy) page.

## Requirements

This page continues the workflows outlined in the [configure storage bucket policies](/boundary/docs/configuration/session-recording/configure-storage-policy) page. The requirements outlined in that page are prerequisites for the workflow defined below.

To apply an updated storage policy to existing session recordings, you first need:

1. An external storage provider configured to store session recordings.
1. A Boundary worker configured for local storage.
1. A Boundary storage bucket with a defined retention and/or deletion policy.
1. A scope with the storage policy attached.
1. A set of session recordings made when the storage policy was attached.

The policies mentioned here demonstrate how to apply an updated storage policy to an existing set of [session recordings][].

## Storage policy changes

A storage policy defines how long the recording within a scope should retain its session recordings. 

> Over time, storage policies may be updated to reflect new organizational requirements, compliance changes, or cost management strategies. While updated policies automatically apply to new session recordings within the scopes associated with that policy, existing recordings maintain the previous policies unless the new policy is directly applied to those recordings.

Reviewer: ```suggestion
Over time, you may update storage policies to reflect new organizational requirements, compliance changes, or cost management strategies. While updated policies automatically apply to new session recordings within the scopes associated with that policy, existing recordings maintain the previous policies unless you apply the new policy directly to those recordings.
```

  "description": "SOC 2 compliant storage policy for session recordings, V2",
  "created_time": "2024-01-25T22:05:52.112677Z",
  "updated_time": "2024-01-30T22:44:08.890715Z",
  "type": "storage",
  "version": 2,
  "attributes": {
    "delete_after": {
      "days": 2757,
      "overridable": false
    },
    "retain_for": {
      "days": 2557,
      "overridable": false
    }
  },
  "authorized_actions": [
    "no-op",
    "read",
    "update",
    "delete"
  ]
}
```

</CodeBlockConfig>

</Tab>
</Tabs>

This policy was applied to the `prod-databases` org in the [configure storage bucket policies](/boundary/docs/configuration/session-recording/configure-storage-policy) page.

New recordings within the `prod-databases` org will automatically have the updated version of the `soc2-policy` applied.

### Reapply a storage policy

Many organizations compliance standards will require that previous versions of storage policies remain applied to existing recordings.

<Note>

  During the initial migration to a Boundary version that includes storage policies, all existing session recordings will have a `retain_for` attribute of `-1`, which retains the recording forever.

</Note>

> In the event that an updated policy should be retroactively applied to existing session recordings, a storage policy should be reapplied.

Reviewer: ```suggestion
In the event that an updated policy should be retroactively applied to existing session recordings, you must reapply the storage policy.
```
